modules/core/include/opencv2/core/cvdef.h

```diff
@@ -933,7 +933,9 @@ inline hfloat hfloatFromBits(ushort w) {
 #endif
 }
 
+#if !defined(__OPENCV_BUILD) && !(defined __STDCPP_FLOAT16_T__) && !(defined __ARM_NEON)
 typedef hfloat float16_t;
+#endif
```

Comment:
This should be dropped in 5.x. That's to say no float16_t in 5.x even for external user.